src/librbd/ImageCtx.cc

```diff
@@ -114,7 +114,7 @@ struct C_AsyncCallback : public Context {
     : image_ctx(image_ctx), on_finish(on_finish) {
   }
   virtual void finish(int r) {
-    image_ctx->op_work_queue->queue(on_finish, r);
+    image_ctx->writeback_handler->queue(on_finish, r);
```

Comment:
@chenyupeng360 The ```flush_aysnc_operations()``` and ```flush_async_operations(Context*)``` methods could be modified to be like [1] where it uses a helper ```Context``` w/ a ```reentrant_safe``` boolean flag. The synchronous (reentrant safe) version can just directly complete.

[1] https://github.com/ceph/ceph/blob/master/src/librbd/ImageCtx.cc#L123